release/models/acl/openconfig-packet-match.yang

```diff
@@ -500,4 +524,110 @@ module openconfig-packet-match {
     }
   }
 
+  grouping ip-icmpv4-header-fields-top {
+    description
+      "Top grouping for ICMPv4 filtering";
+
+    container icmpv4 {
+      description
+        "Top container for ICMPv4 filtering";
+
+      container config {
+        description
+          "Configuration attributes for ICMPv4 filtering";
+
+        uses ip-icmpv4-header-fields-config;
+      }
+
+      container state {
+        description
+          "State attributes for ICMPv4 filtering";
+        config false;
+
+        uses ip-icmpv4-header-fields-config;
+      }
+    }
+  }
+
+  grouping ip-icmpv6-header-fields-top {
+    description
+      "Top grouping for ICMPv6 filtering";
+
+    container icmpv6 {
+      description
+        "Top container for ICMPv6 filtering";
+
+      container config {
+        description
+          "Configuration attributes for ICMPv6 filtering";
+
+        uses ip-icmpv6-header-fields-config;
+      }
+
+      container state {
+        description
+          "State attributes for ICMPv6 filtering";
+        config false;
+
+        uses ip-icmpv6-header-fields-config;
+      }
+    }
+  }
+
+  grouping ip-icmpv4-header-fields-config {
+    description
+      "Collection of ICMPv4 header fields that can be
+       used to set up a match filter.";
+
+    leaf type {
+      type identityref {
```

Comment:
Would it be better to make it an union as in the case of the protocol match? either a number or the identify?

Reply:
I pondered doing so, the main reason I decided against this was that this just gave two ways to express the same thing. The only benefit of doing so is that you can express something that we didn't add as an identity here, but given that we've been somewhat exhaustive, it's not clear to me that we need both.

Let's discuss tomorrow.

Reply:
I get your point. I was reviewing our ACLs requirements and found gaps in the protocol match identities, but as a number is also allowed, there was no urgent issue… So it immediately reminded me of the icmp type/ code matches…

I doubt that there is going to appear in the near future a value that is not in the exhaustive list and someone needs to match… 

On the other hand, the identities need maintenance sooner or later (in IETF there is an ongoing debate on IANA maintained modules that keep track of identities in parallel to number assignments)

By allowing the number too… you can take out the “urgency” to assign new identities to new type/codes when they appear.

Reply:
Agreed -- but equally, I think these things are relatively slow to be added -- an implementation can always define an identity with the same `base` to be able to extend these to different values. My preference is to keep this as a simple identity.

Reply:
I'm ok with identity only.